src/Middleware/Authorization.ts

```diff
@@ -13,8 +13,10 @@ export async function AuthorizeUser(
 ) {
     try {
         let token = (req.headers.authorization || req.headers.Authorization) as string;
-        if (token.startsWith('Bearer ')) {
+        if (token?.startsWith('Bearer ')) {
             token = token.slice(7, token.length).trimLeft();
+        } else {
+            throw new ServiceError(400, 'Bad request.')
```

Comment:
Not really a good solution as you just proposed that token MUST start with `Bearer `. We have been loose on that standard, on frontend we are not even sending that I think.

I think the easy solution would be to check if token exists before this condition.

Reply:
Should I then omit the `Bearer token` check and just check `if(!token)`?

Reply:
This is not the check only. This cleans the string from`Bearer ` if it's sent and only uses token later